Frontend/src/packages/dashboard/api/helpers.ts

```diff
@@ -1,14 +1,24 @@
-import { FileEntity } from "../state/folders/types";
+import {FileEntity, sliceState} from "../state/folders/types";
 import { JSONFileFormat } from "./types";
+import {useSelector} from "react-redux";
+import {RootState} from "../../../redux-state/reducers";
+import {folderSelectors} from "../state/folders";
 
 // Converts a backend response to the File or Folder type
 export function toFileOrFolder(json: JSONFileFormat): FileEntity {
   console.log(json)
-  const {EntityID, EntityName, IsDocument} = json;
+  const {EntityID, EntityName, IsDocument, Parent} = json;
 
   return {
     id: EntityID,
     name: EntityName,
+    parentId: Parent,
     type: IsDocument ? "File" : "Folder",
   } as FileEntity
+}
+
+export function getFolderState(): sliceState  {
+  return useSelector((state: RootState) => (
+    folderSelectors.getFolderState(state)
+  ));
```

Comment:
this can be written as 
```suggestion
export const getFolderState = useSelector(
  (state: RootState) => (folderSelectors.getFolderState(state))
);
```
which can potentially improve readability due to there not being a function call being nested in the return value

Reply:
This is necessary because without it being nested, react will raise an error for an invalid hook call.